exercises/two-bucket/.meta/src/reference/java/TwoBucket.java

```diff
@@ -0,0 +1,139 @@
+import java.util.ArrayList;
+
+class TwoBucket {
+	
+    private class State {
+    	int moves;
+    	int bucketOne;
+    	int bucketTwo;
+    	
+    	public State (int moves, int bucketOne, int bucketTwo) {
+    		this.moves = moves;
+    		this.bucketOne = bucketOne;
+    		this.bucketTwo = bucketTwo;
+    	}
+    	
+    	@Override
+    	public boolean equals(Object o) {
+    		State otherState = (State)o;
+    		return this.moves == otherState.moves &&
+    				this.bucketOne == otherState.bucketOne &&
+    				this.bucketTwo == otherState.bucketTwo;
+    	}
+    }
+    
+    private State finalState;
+    
+    int bucketOneCap;
+    int bucketTwoCap;
+    int desiredLiters;
+    String first;
+    
+    public TwoBucket(int bucketOneCap, int bucketTwoCap, int desiredLiters, String first) {
+    	this.bucketOneCap = bucketOneCap;
+    	this.bucketTwoCap = bucketTwoCap;
+    	this.desiredLiters = desiredLiters;
+    	this.first = first;
+    	
+    	finalState = computeFinalState();
+    }
+    
+    private ArrayList<State> getAdjacentStates (State state) {
+    	ArrayList<State> adjacentStates = new ArrayList<State>();
+    	
+    	//Empty bucket one
+    	adjacentStates.add(new State(state.moves + 1, 0, state.bucketTwo));
+    	
+    	//Empty bucket two
+    	adjacentStates.add(new State(state.moves + 1, state.bucketOne, 0));
+    	
+    	//Fill bucket one
+    	adjacentStates.add(new State(state.moves + 1, bucketOneCap, state.bucketTwo));
+    	
+    	//Fill bucket two
+    	adjacentStates.add(new State(state.moves + 1, state.bucketOne, bucketTwoCap));
+    	
+    	//pour from bucket one to bucket two
+    	if(state.bucketOne + state.bucketTwo > bucketTwoCap) {
```

Comment:
Minor nitpick: there should be a whitespace after `if` in this and the other `if` statements 🙂